packages/core/src/test/awsService/accessanalyzer/iamPolicyChecks.test.ts

```diff
@@ -0,0 +1,793 @@
+import { FakeExtensionContext } from '../../../test/fakeExtensionContext'
+import fs from 'fs'
+import { VueWebviewPanel } from '../../../webviews/main'
+import { globals } from '../../../shared'
+import { AccessAnalyzer } from 'aws-sdk'
+import * as s3Client from '../../../shared/clients/s3Client'
+import { DefaultS3Client } from '../../../shared/clients/s3Client'
+import * as iamPolicyChecks from '../../../awsService/accessanalyzer/vue/iamPolicyChecks'
+import * as vscode from 'vscode'
+import { IamPolicyChecksConstants } from '../../../awsService/accessanalyzer/vue/constants'
+
+let policyChecksWebview: VueWebviewPanel | undefined
+const defaultTerraformConfigPath = 'resources/policychecks-tf-default.yaml'
+let sandbox: sinon.SinonSandbox
+let fakePolicyChecksWebview: IamPolicyChecksWebview
+let mockAccessAnalyzer: sinon.SinonMock
+
+describe('iamPolicyChecks', function () {
+    before(async function () {
+        policyChecksWebview = await renderIamPolicyChecks(await FakeExtensionContext.getFakeExtContext())
+    })
+
+    it('webview is not undefined', async function () {
+        assert(policyChecksWebview !== undefined)
+    })
+
+    it('IamPolicyChecksWebview built .vue source file path exists', async function () {
+        assert.ok(fs.existsSync(globals.context.asAbsolutePath(`dist/${policyChecksWebview?.server.source}`)))
+    })
+
+    it('default Terraform config exists', async function () {
+        assert.ok(fs.existsSync(globals.context.asAbsolutePath(defaultTerraformConfigPath)))
+    })
+})
+
+describe('_readCustomChecksFile', () => {
+    let fsExistsSyncStub: sinon.SinonStub
+    let fsReadFileSyncStub: sinon.SinonStub
+    let parseS3UriStub: sinon.SinonStub
+    let s3ClientStub: sinon.SinonStubbedInstance<DefaultS3Client>
+    sandbox = sinon.createSandbox()
+
+    beforeEach(() => {
+        fsExistsSyncStub = sandbox.stub(fs, 'existsSync')
+        fsReadFileSyncStub = sandbox.stub(fs, 'readFileSync')
+        parseS3UriStub = sandbox.stub(s3Client, 'parseS3Uri')
+        s3ClientStub = sandbox.createStubInstance(DefaultS3Client)
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    it('should read file content if file exists', async () => {
+        const filePath = 'path/to/file.txt'
+        const fileContent = 'file content'
+
+        fsExistsSyncStub.withArgs(filePath).returns(true)
+        fsReadFileSyncStub.withArgs(filePath).returns(fileContent)
+
+        const result = await _readCustomChecksFile(filePath)
+
+        assert.strictEqual(result, fileContent)
+        assert(fsExistsSyncStub.calledWith(filePath))
+        assert(fsReadFileSyncStub.calledWith(filePath))
+    })
+
+    it('should throw PolicyChecksError for invalid S3 URI', async () => {
+        const invalidS3Uri = 's3://invalid-uri'
+
+        fsExistsSyncStub.withArgs(invalidS3Uri).returns(false)
+        parseS3UriStub.withArgs(invalidS3Uri).throws(new Error('Invalid S3 URI'))
```

Comment:
why is this stubbed? does it depend on the AWS service? does it make network calls?

avoid stubs/mocks, they are fragile/brittle and lead to tests that do very little.

https://github.com/aws/aws-toolkit-vscode/blob/master/docs/CODE_GUIDELINES.md#test-guidelines

Reply:
This does not make a call to the AWS service, I can remove the stub here.
Q: I have a mock for our AccessAnalyzer SDK client, is that ok? It looks like the checks are failing because of it:
```
Error: ENOENT: no such file or directory, open '/Users/runner/.aws/config'
```
Looks like the initialization of the client is trying to read that directory; is there a better way I should go about this when working with AWS clients?